### PR TITLE
case 20058: Android - Handshake implementation

### PR DESCRIPTION
**Pull request by Cristo86**
https://highfidelity.manuscript.com/f/cases/20058/User-on-mobile-unable-to-handshake

Handshake implementation:

**Test plan**

1. Login with two users, only one from Android is necessary for testing.
2. Preconditions required for Desktop handshake are also required on Android.
3. Press the right button with a handshake icon to start the process, avatars should be in front of each other. Avatars raise the hand as in Desktop.
4. After completing the process, both avatars should be connected.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12486/HighFidelity-Beta-PR13901-4f3a730.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12486/HighFidelity-Beta-PR13901-4f3a730.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12486/HighFidelity-Beta-Interface-PR13901-4f3a730.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12486/HighFidelity-Beta-PR13901-4f3a730.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12486/HighFidelity-Beta-Interface-PR13901-4f3a730.exe

File: libraries/animation/src/AnimInverseKinematics.cpp [hunk 1]
```diff
@@ -851,10 +851,12 @@ const AnimPoseVec& AnimInverseKinematics::evaluate(const AnimVariantMap& animVar
 
 //virtual
 const AnimPoseVec& AnimInverseKinematics::overlay(const AnimVariantMap& animVars, const AnimContext& context, float dt, AnimVariantMap& triggersOut, const AnimPoseVec& underPoses) {
+/* We need this working so body parts can move
 #ifdef Q_OS_ANDROID
     // disable IK on android
     return underPoses;
 #endif
+*/
```
Comment (SamGondelman) [inline libraries/animation/src/AnimInverseKinematics.cpp]: If we're re-enabling this, we should be able to at least make it less accurate.  @hyperlogic might have a better suggestion.

**Comment by Cristo86** [comment]
build please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12600/HighFidelity-Beta-PR13901-a6c1f3e.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12600/HighFidelity-Beta-PR13901-a6c1f3e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12600/HighFidelity-Beta-Interface-PR13901-a6c1f3e.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12602/HighFidelity-Beta-PR13901-2e43912.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12602/HighFidelity-Beta-PR13901-2e43912.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12602/HighFidelity-Beta-Interface-PR13901-2e43912.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12612/HighFidelity-Beta-PR13901-8839a3b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12612/HighFidelity-Beta-PR13901-8839a3b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12612/HighFidelity-Beta-Interface-PR13901-8839a3b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12612/HighFidelity-Beta-PR13901-8839a3b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12612/HighFidelity-Beta-Interface-PR13901-8839a3b.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16948/HighFidelity-Beta-PR13901-6a66583.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16948/HighFidelity-Beta-PR13901-6a66583.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16948/HighFidelity-Beta-Interface-PR13901-6a66583.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17113/HighFidelity-Beta-PR13901-babb972.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17113/HighFidelity-Beta-PR13901-babb972.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17113/HighFidelity-Beta-Interface-PR13901-babb972.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17113/HighFidelity-Beta-PR13901-babb972.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17113/HighFidelity-Beta-Interface-PR13901-babb972.exe

**Comment by Cristo86** [comment]
This PR should be tested after https://github.com/highfidelity/hifi/pull/14865 is merged, and if the performance is ok, just remove that block (AnimInverseKinematics.cpp) instead of commenting it out.

File: libraries/animation/src/AnimInverseKinematics.cpp (hunk 1 above)
Comment (hyperlogic) [reply libraries/animation/src/AnimInverseKinematics.cpp]: Re-enabling this should be fine.  For non-hmd use cases, IK is disabled completely.  However, the handshake script does use IK to raise the hand.  In the short term there will be a performance hit for this.  However we have several optimizations that will land in master in the next few weeks that should improve IK performance by as much as 10x.  Such as #14817

Comment (SamGondelman) [reply libraries/animation/src/AnimInverseKinematics.cpp]: Ok, then just delete this commented out section completely @Cristo86

Comment (Cristo86) [reply libraries/animation/src/AnimInverseKinematics.cpp]: Perfect, done.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17165/HighFidelity-Beta-PR13901-514c552.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17165/HighFidelity-Beta-PR13901-514c552.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17165/HighFidelity-Beta-Interface-PR13901-514c552.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17165/HighFidelity-Beta-PR13901-514c552.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17165/HighFidelity-Beta-Interface-PR13901-514c552.exe

**Comment by samcake** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17460/HighFidelity-Beta-PR13901-514c552.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17460/HighFidelity-Beta-PR13901-514c552.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17460/HighFidelity-Beta-Interface-PR13901-514c552.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17460/HighFidelity-Beta-PR13901-514c552.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17460/HighFidelity-Beta-Interface-PR13901-514c552.exe

**Pull request merged**